_posts/2025-01-16-19.0.0-release.md

```diff
@@ -0,0 +1,239 @@
+  dtype](https://pandas.pydata.org/pdeps/0014-string-dtype.html) is now
+  supported by PyArrow's `to_pandas` routine. In the future, when using pandas >=3.0,
+  the new pandas behavior will be enabled by default. You can opt into
+  the new behavior under pandas >=2.3 by setting `pd.options.future.infer_string
+  = True`. This may be considered a breaking change.
+  ([#43683](https://github.com/apache/arrow/issues/43683))
+- Support for 32-bit and 64-bit decimal types was added.
+  ([#44713](https://github.com/apache/arrow/issues/44713))
+- Arrow PyCapsule stream objects are supported in `write_dataset`.
+  ([#43410](https://github.com/apache/arrow/issues/43410))
+- New Flight features have been exposed.
+  ([#36954](https://github.com/apache/arrow/issues/36954))
+- Bindings for `JsonExtensionType` and `JsonArray` were added.
+  ([#44066](https://github.com/apache/arrow/issues/44066))
+- Hyperbolic trigonometry functions added to the Arrow C++ compute kernels are
+  also available in PyArrow.
+  ([#44952](https://github.com/apache/arrow/issues/44952))
+
+Other improvements:
+
+- `strings_to_categorical` keyword in `to_pandas` can now be used for string
+  view type. ([#45175](https://github.com/apache/arrow/issues/45175))
+- `from_buffers` is updated to work with `StringView`.
+  ([#44651](https://github.com/apache/arrow/issues/44651))
+- Version suffixes are also set for Arrow Python C++ (`libarrow_python*`)
+  libraries. ([#44614](https://github.com/apache/arrow/issues/44614))
+
+## Ruby and C GLib Notes
+
+### Ruby
+
+- Added basic support for JRuby with an implementation based on Arrow Java
+  ([#44346](https://github.com/apache/arrow/pull/44346)). The plan is to release
+  this as a gem once it covers a base set of features. See
+  [#45324](https://github.com/apache/arrow/issues/45324) for more information.
+- Added support for 32bit and 64bit decimal, binary view, and string view. See
+  [issues
+  listed](https://github.com/apache/arrow/issues?q=is%3Aclosed%20milestone%3A19.0.0%20label%3A%22Component%3A%20GLib%22)
+  in the 19.0.0 milestone for more details.
+- Fixed a bug that empty struct list can't be built.
+  ([#44742](https://github.com/apache/arrow/issues/44742))
+- Fixed a bug that `record_batch[:column].size` raises an exception.
+  ([#45119](https://github.com/apache/arrow/issues/45119))
+
+### C GLib
+
+- Added support for 32bit and 64bit decimal, binary view, and string view. See
+  [issues listed in the 19.0.0
+  milestone](https://github.com/apache/arrow/issues?q=is%3Aclosed%20milestone%3A19.0.0%20label%3A%22Component%3A%20GLib%22)
+  for more details.
+
+[1]: https://github.com/apache/arrow/milestone/66?closed=1
+[2]: {{ site.baseurl }}/release/19.0.0.html#contributors
+[3]: {{ site.baseurl }}/release/19.0.0.html#changelog
+[4]: {{ site.baseurl }}/docs/r/news/
+[5]: https://github.com/apache/arrow-rs/tags
+[6]: https://github.com/apache/arrow-go/tags
+[7]: https://github.com/apache/arrow-java/tags
```

Comment:
"releases" may be better than "tags" because "releases" have notes associated with tags.

```suggestion
[5]: https://github.com/apache/arrow-rs/releases
[6]: https://github.com/apache/arrow-go/releases
[7]: https://github.com/apache/arrow-java/releases
```

Reply:
Good point. Though it looks like only arrow-go is using releases like we'd expect. Java and Rust aren't using GitHub releases. Let me find better links for java and Rust.

Reply:
Java will use releases. So we can use releases for Java.

Reply:
Noted, thanks. For Rust, we can link to their CHANGELOG.md I think, https://github.com/apache/arrow-rs/blob/main/CHANGELOG.md.

Reply:
+1

@alamb It may be better that arrow-rs creates GibHub Releases (https://github.com/apache/arrow-rs/releases) with the generated changelog. FYI: We can create a GitHub Releases by `gh release create ${GITHUB_REF_NAME} --notes-file CHANGELOG.md` (with `env: GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}`) by GItHub Actions automatically. See also https://github.com/apache/arrow-java/blob/main/.github/workflows/release.yml .

Reply:
Thanks @kou - I will try it with the next release. Tracked by
- https://github.com/apache/arrow-rs/issues/6929#issuecomment-2615268539